Add `frozen_string_literal: true` to the top of all files & fix issues with frozen strings

parkr opened the pull request:
In #289, @MikeMcQuaid suggested that we test the entire project with `--enable-frozen-string-literal`. I gave that a shot and unfortunately couldn't make it work (RSpec has several string-modifying method calls which rely on string mutability in that library).

In Jekyll, we found that freezing strings greatly sped up the execution of the program since the Ruby VM didn't have to continue allocating new strings every time it saw a literal.

In an effort to bring this project into full compliance with frozen string literals, I submit to you the following. It does not include the fixes in #289, but they could be wrapped into 1 PR if that's what you'd prefer.

/cc @MikeMcQuaid

MikeMcQuaid commented:
Once this is 💚 : I'm game for this!

parkr commented:
Rebased on latest `master`.

parkr commented:
CI runs Ruby 2.0, so I went back to using `<<-EOS.undent`.

parkr commented:
@MikeMcQuaid Tests passed! Unfortunately, something else didn't, which is why we have 🔴 CI.

```text
Finished in 11.54 seconds (files took 0.8371 seconds to load)

[32m116 examples, 0 failures[0m



Coverage report generated for RSpec to /Users/travis/build/Homebrew/homebrew-bundle/coverage. 1369 / 1369 LOC (100.0%) covered.



travis_time:end:1c546ff4:start=1507041692628521000,finish=1507041706790379000,duration=14161858000
[0K

[32;1mThe command "bundle exec rake" exited with 0.[0m

travis_fold:start:cache.2
[0Kstore build cache

$ bundle clean

travis_time:start:02c45112
[0K

travis_time:end:02c45112:start=1507041707124640000,finish=1507041707139957000,duration=15317000
[0Ktravis_time:start:12b0ce7a
[0K[32;1mnothing changed, not updating cache[0m



travis_time:end:12b0ce7a:start=1507041707174828000,finish=1507041717426156000,duration=10251328000
[0Ktravis_fold:end:cache.2
[0K

Done. Your build exited with 1.

/Users/travis/.travis/job_stages: line 1
```

MikeMcQuaid commented:
> CI runs Ruby 2.0, so I went back to using <<-EOS.undent.

@parkr This shouldn't be the case any more as-of https://github.com/Homebrew/homebrew-bundle/pull/292 (now merged).

parkr commented:
>>  CI runs Ruby 2.0, so I went back to using <<-EOS.undent.
>
> @parkr This shouldn't be the case any more as-of #292 (now merged).

Huzzah! Rebased and updated.

parkr commented:
@MikeMcQuaid Tests are passing! There is an issue running `brew style homebrew/homebrew-bundle` in CI, however, since it's targeting Ruby 2.0. The fix for this is to target Ruby 2.3 in the rubocop file that `brew style` uses.

```text
== /usr/local/Homebrew/Library/Taps/homebrew/homebrew-bundle/lib/bundle/brew_dumper.rb ==

E:176: 87: unexpected token tBANG

(Using Ruby 2.0 parser; configure using TargetRubyVersion parameter, under AllCops)

== /usr/local/Homebrew/Library/Taps/homebrew/homebrew-bundle/lib/bundle/brew_installer.rb ==

C:158: 13: Inconsistent indentation detected.

W:158: 13: Variable Unlinking used in void context.

C:158: 23: Missing space after #.

C:159: 13: Inconsistent indentation detected.

C:159: 39: Use && instead of and.

C:159: 58: Missing space after #.

== /usr/local/Homebrew/Library/Taps/homebrew/homebrew-bundle/spec/brew_services_spec.rb ==

C: 19:  9: Inconsistent indentation detected.

C: 20:  9: Inconsistent indentation detected.

C: 21:  9: Inconsistent indentation detected.

W: 22:  7: Variable EOS used in void context.
```

MikeMcQuaid commented:
Thanks @parkr!

MikeMcQuaid commented:
(I reran CI after a Homebrew/brew change to `brew style` in the new tag)

parkr commented:
Brilliant! Thanks, Mike!
